pkg/api/configsync/v1alpha1/resource_override.go

```diff
@@ -170,3 +174,14 @@ type ContainerLogLevelOverride struct {
 	// +kubebuilder:validation:Required
 	LogLevel int `json:"logLevel"`
 }
+
+// GitSyncOverride specifies the git-sync container override value
+type GitSyncOverride struct {
+	// gitSyncTimeout allows one to configure the timeout for the git-sync container.
+	// Must be no less than 0.
+	// If this field is not provided, the default value is 2 minutes.
+	//
+	// +kubebuilder:validation:Minimum=0
+	// +optional
+	GitSyncTimeout int `json:"gitSyncTimeout,omitempty"`
```

Comment:
`gitSyncOverride.gitSyncTimeout` seems a bit redundant. I think this could just be `timeout`.